### PR TITLE
Process pm

### PR DESCRIPTION
**Pull request by Ruifang-Li**
This is program to read in PM observation from ASCII file and convert to PrepBUFR format for PM DA.
Test with real time runs on Hera is successful.

### CODE REVIEW

File: pm/CMakeLists.txt [hunk 1]
```diff
@@ -0,0 +1,19 @@
+list(APPEND src_pm
+  process_pm.f90)
+
+add_executable(process_pm.exe ${src_pm})
+target_link_libraries(process_pm.exe PRIVATE ${PROJECT_NAME}::ncio
+                                          PRIVATE ${PROJECT_NAME}::esggrid_util
+                                          PRIVATE ${PROJECT_NAME}::pesglib
+                                          PRIVATE NetCDF::NetCDF_Fortran
+                                          PRIVATE gsi::gsi
```
Comment (hu5970) [inline pm/CMakeLists.txt]: May not need ncio,esggird_util,pesglib, gsi

Comment (hu5970) [reply pm/CMakeLists.txt]: You can delete them and try again.

File: pm/process_pm.f90 [hunk 2]
```diff
@@ -0,0 +1,143 @@
...
+ allocate(pm_str(cnt))
+
+ open (unit = unit_var, file = infile,status = 'old')
+ do i = 1,cnt
+   read(unit_var,'(7a16)')            sta_id(i), lat_str(i), lon_str(i), elv_str(i), pm_aqi_str(i), pm_measured_str(i), pm_str(i)
+   !write(*,'(a10,7a15)')     'obs  ', trim(adjustl(sta_id(i))), trim(adjustl(lat_str(i))), trim(adjustl(lon_str(i))),trim(adjustl(elv_str(i))), &
+   !                                   trim(adjustl(pm_aqi_str(i))), trim(adjustl(pm_measured_str(i))), trim(adjustl(pm_str(i)))
+   !write(*,'(a10,7i15)')     'len  ', len(trim(adjustl(sta_id(i)))), len(trim(adjustl(lat_str(i)))), len(trim(adjustl(lon_str(i)))), len(trim(adjustl(elv_str(i)))), &
+   !                                   len(trim(adjustl(pm_aqi_str(i)))), len(trim(adjustl(pm_measured_str(i)))), len(trim(adjustl(pm_str(i))))
+ end do
```
Comment (hu5970) [inline pm/process_pm.f90:61]: remove line 61-64  if they are not needed.

File: pm/process_pm.f90 [hunk 3]
```diff
@@ -0,0 +1,143 @@
...
+ !call maxout(20000) ! 20k byte any message size
+ call datelen(10)
+
+ ! unchaged variables for each profile at current time
+ subset="ANOWPM"
+ read(analysis_time,'(I10)')  idate    ! string to integer
+
+ ! hdstr='SID XOB YOB DHR TYP T29 SQN PROCN RPT CAT TYPO TSIG'
+ ! obstr='TPHR QCIND COPOPM'
+
```
Comment (hu5970) [inline pm/process_pm.f90:79]: remove line 76, 83,84 if they are not needed.

File: pm/process_pm.f90 [hunk 4]
```diff
@@ -0,0 +1,143 @@
...
+    str=adjustl(elv_str(j))
+    read(str(2:),'(f15.6)') elv
+
+    str=adjustl(pm_str(j))
+    read(str(2:),'(f15.6)') pm
+
+    str=adjustl(sta_id(j))
+    c_sid=str(2:9)
+    !write(*,'(2a15,4f15.5)')     'Obs   ', c_sid, lat, lon, elv, pm 
+    hdr(1)=rstation_id; hdr(2)=lon; hdr(3)=lat; hdr(4)=0.0_8; hdr(5)=102; hdr(10)=6 ! Single level report
```
Comment (hu5970) [inline pm/process_pm.f90:103]: remove line 109

File: pm/CMakeLists.txt (hunk 1 above)
Comment (Ruifang-Li) [reply pm/CMakeLists.txt]: done

File: pm/process_pm.f90 (hunk 2 above)
Comment (Ruifang-Li) [reply pm/process_pm.f90:61]: done

File: pm/process_pm.f90 (hunk 3 above)
Comment (Ruifang-Li) [reply pm/process_pm.f90:79]: done

File: pm/process_pm.f90 (hunk 4 above)
Comment (Ruifang-Li) [reply pm/process_pm.f90:103]: done

**Review by hu5970 — APPROVED** [review]

**Review by guoqing-noaa — APPROVED** [review]

**Pull request merged**